src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -102,10 +109,11 @@ CHIP_ERROR BDXDownloader::SetBDXParams(const chip::bdx::TransferSession::Transfe
 
     VerifyOrReturnError(mState == State::kIdle, CHIP_ERROR_INCORRECT_STATE);
 
+    mBdxTransfer.RegisterOutputEventCallback(this, OnBDXEventReceived);
```

Comment:
while reviewing right now I realized that the callback should be set when we PrepareforTransfer or Initiate transfer from requestor so maybe add a param to those API's rather than have a seperate register/unregister because its kind of required to have a callback for BDX to work. It seems more error prone to have a separate register/unregister. i will change this. Please lmk wdyt?